File: qt/gen_qpixelformat.cpp
// +build ignore

#include <QPixelFormat>
#include <qpixelformat.h>
#include "gen_qpixelformat.h"

#ifndef _Bool
#define _Bool bool
#endif

QPixelFormat* QPixelFormat_new() {
	return new QPixelFormat();
}

QPixelFormat* QPixelFormat_new2(ColorModel colorModel, unsigned char firstSize, unsigned char secondSize, unsigned char thirdSize, unsigned char fourthSize, unsigned char fifthSize, unsigned char alphaSize, AlphaUsage alphaUsage, AlphaPosition alphaPosition, AlphaPremultiplied premultiplied, TypeInterpretation typeInterpretation) {
	return new QPixelFormat(colorModel, static_cast<uchar>(firstSize), static_cast<uchar>(secondSize), static_cast<uchar>(thirdSize), static_cast<uchar>(fourthSize), static_cast<uchar>(fifthSize), static_cast<uchar>(alphaSize), alphaUsage, alphaPosition, premultiplied, typeInterpretation);
}

QPixelFormat* QPixelFormat_new3(QPixelFormat* param1) {
	return new QPixelFormat(*param1);
}

QPixelFormat* QPixelFormat_new4(ColorModel colorModel, unsigned char firstSize, unsigned char secondSize, unsigned char thirdSize, unsigned char fourthSize, unsigned char fifthSize, unsigned char alphaSize, AlphaUsage alphaUsage, AlphaPosition alphaPosition, AlphaPremultiplied premultiplied, TypeInterpretation typeInterpretation, ByteOrder byteOrder) {
	return new QPixelFormat(colorModel, static_cast<uchar>(firstSize), static_cast<uchar>(secondSize), static_cast<uchar>(thirdSize), static_cast<uchar>(fourthSize), static_cast<uchar>(fifthSize), static_cast<uchar>(alphaSize), alphaUsage, alphaPosition, premultiplied, typeInterpretation, byteOrder);
}

QPixelFormat* QPixelFormat_new5(ColorModel colorModel, unsigned char firstSize, unsigned char secondSize, unsigned char thirdSize, unsigned char fourthSize, unsigned char fifthSize, unsigned char alphaSize, AlphaUsage alphaUsage, AlphaPosition alphaPosition, AlphaPremultiplied premultiplied, TypeInterpretation typeInterpretation, ByteOrder byteOrder, unsigned char subEnum) {
	return new QPixelFormat(colorModel, static_cast<uchar>(firstSize), static_cast<uchar>(secondSize), static_cast<uchar>(thirdSize), static_cast<uchar>(fourthSize), static_cast<uchar>(fifthSize), static_cast<uchar>(alphaSize), alphaUsage, alphaPosition, premultiplied, typeInterpretation, byteOrder, static_cast<uchar>(subEnum));
}

ColorModel QPixelFormat_ColorModel(const QPixelFormat* self) {
	return self->colorModel();
}

unsigned char QPixelFormat_ChannelCount(const QPixelFormat* self) {
	uchar _ret = self->channelCount();
	return static_cast<unsigned char>(_ret);
}

unsigned char QPixelFormat_RedSize(const QPixelFormat* self) {
	uchar _ret = self->redSize();
	return static_cast<unsigned char>(_ret);
}

unsigned char QPixelFormat_GreenSize(const QPixelFormat* self) {
	uchar _ret = self->greenSize();
	return static_cast<unsigned char>(_ret);
}

unsigned char QPixelFormat_BlueSize(const QPixelFormat* self) {
	uchar _ret = self->blueSize();
	return static_cast<unsigned char>(_ret);
}

unsigned char QPixelFormat_CyanSize(const QPixelFormat* self) {
	uchar _ret = self->cyanSize();
	return static_cast<unsigned char>(_ret);
}

unsigned char QPixelFormat_MagentaSize(const QPixelFormat* self) {
	uchar _ret = self->magentaSize();
	return static_cast<unsigned char>(_ret);
}

unsigned char QPixelFormat_YellowSize(const QPixelFormat* self) {
	uchar _ret = self->yellowSize();
	return static_cast<unsigned char>(_ret);
}

unsigned char QPixelFormat_BlackSize(const QPixelFormat* self) {
	uchar _ret = self->blackSize();
	return static_cast<unsigned char>(_ret);
}

unsigned char QPixelFormat_HueSize(const QPixelFormat* self) {
	uchar _ret = self->hueSize();
	return static_cast<unsigned char>(_ret);
}

unsigned char QPixelFormat_SaturationSize(const QPixelFormat* self) {
	uchar _ret = self->saturationSize();
	return static_cast<unsigned char>(_ret);
}

unsigned char QPixelFormat_LightnessSize(const QPixelFormat* self) {
	uchar _ret = self->lightnessSize();
	return static_cast<unsigned char>(_ret);
}

unsigned char QPixelFormat_BrightnessSize(const QPixelFormat* self) {
	uchar _ret = self->brightnessSize();
	return static_cast<unsigned char>(_ret);
}

unsigned char QPixelFormat_AlphaSize(const QPixelFormat* self) {
	uchar _ret = self->alphaSize();
	return static_cast<unsigned char>(_ret);
}

unsigned char QPixelFormat_BitsPerPixel(const QPixelFormat* self) {
	uchar _ret = self->bitsPerPixel();
	return static_cast<unsigned char>(_ret);
}

AlphaUsage QPixelFormat_AlphaUsage(const QPixelFormat* self) {
	return self->alphaUsage();
}

AlphaPosition QPixelFormat_AlphaPosition(const QPixelFormat* self) {
	return self->alphaPosition();
}

AlphaPremultiplied QPixelFormat_Premultiplied(const QPixelFormat* self) {
	return self->premultiplied();
}

TypeInterpretation QPixelFormat_TypeInterpretation(const QPixelFormat* self) {
	return self->typeInterpretation();
}

ByteOrder QPixelFormat_ByteOrder(const QPixelFormat* self) {
	return self->byteOrder();
}

YUVLayout QPixelFormat_YuvLayout(const QPixelFormat* self) {
	return self->yuvLayout();
}

unsigned char QPixelFormat_SubEnum(const QPixelFormat* self) {
	uchar _ret = self->subEnum();
	return static_cast<unsigned char>(_ret);
}

void QPixelFormat_Delete(QPixelFormat* self, bool isSubclass) {
	if (isSubclass) {
		delete dynamic_cast<QPixelFormat*>( self );
	} else {
		delete self;
	}
}

File: qt/gen_qpixelformat.go
package qt

import (
	"unsafe"
)

type QPixelFormat__ColorModel int

const (
	QPixelFormat__RGB       QPixelFormat__ColorModel = 0
	QPixelFormat__BGR       QPixelFormat__ColorModel = 1
	QPixelFormat__Indexed   QPixelFormat__ColorModel = 2
	QPixelFormat__Grayscale QPixelFormat__ColorModel = 3
	QPixelFormat__CMYK      QPixelFormat__ColorModel = 4
	QPixelFormat__HSL       QPixelFormat__ColorModel = 5
	QPixelFormat__HSV       QPixelFormat__ColorModel = 6
	QPixelFormat__YUV       QPixelFormat__ColorModel = 7
	QPixelFormat__Alpha     QPixelFormat__ColorModel = 8
)

type QPixelFormat__AlphaUsage int

const (
	QPixelFormat__UsesAlpha    QPixelFormat__AlphaUsage = 0
	QPixelFormat__IgnoresAlpha QPixelFormat__AlphaUsage = 1
)

type QPixelFormat__AlphaPosition int

const (
	QPixelFormat__AtBeginning QPixelFormat__AlphaPosition = 0
	QPixelFormat__AtEnd       QPixelFormat__AlphaPosition = 1
)

type QPixelFormat__AlphaPremultiplied int

const (
	QPixelFormat__NotPremultiplied QPixelFormat__AlphaPremultiplied = 0
	QPixelFormat__Premultiplied    QPixelFormat__AlphaPremultiplied = 1
)

type QPixelFormat__TypeInterpretation int

const (
	QPixelFormat__UnsignedInteger QPixelFormat__TypeInterpretation = 0
	QPixelFormat__UnsignedShort   QPixelFormat__TypeInterpretation = 1
	QPixelFormat__UnsignedByte    QPixelFormat__TypeInterpretation = 2
	QPixelFormat__FloatingPoint   QPixelFormat__TypeInterpretation = 3
)

type QPixelFormat__YUVLayout int

const (
	QPixelFormat__YUV444   QPixelFormat__YUVLayout = 0
	QPixelFormat__YUV422   QPixelFormat__YUVLayout = 1
	QPixelFormat__YUV411   QPixelFormat__YUVLayout = 2
	QPixelFormat__YUV420P  QPixelFormat__YUVLayout = 3
	QPixelFormat__YUV420SP QPixelFormat__YUVLayout = 4
	QPixelFormat__YV12     QPixelFormat__YUVLayout = 5
	QPixelFormat__UYVY     QPixelFormat__YUVLayout = 6
	QPixelFormat__YUYV     QPixelFormat__YUVLayout = 7
	QPixelFormat__NV12     QPixelFormat__YUVLayout = 8
	QPixelFormat__NV21     QPixelFormat__YUVLayout = 9
	QPixelFormat__IMC1     QPixelFormat__YUVLayout = 10
	QPixelFormat__IMC2     QPixelFormat__YUVLayout = 11
	QPixelFormat__IMC3     QPixelFormat__YUVLayout = 12
	QPixelFormat__IMC4     QPixelFormat__YUVLayout = 13
	QPixelFormat__Y8       QPixelFormat__YUVLayout = 14
	QPixelFormat__Y16      QPixelFormat__YUVLayout = 15
)

type QPixelFormat__ByteOrder int

const (
	QPixelFormat__LittleEndian        QPixelFormat__ByteOrder = 0
	QPixelFormat__BigEndian           QPixelFormat__ByteOrder = 1
	QPixelFormat__CurrentSystemEndian QPixelFormat__ByteOrder = 2
)

type QPixelFormat struct {
	h          uintptr
	isSubclass bool
}

// NewQPixelFormat constructs a new QPixelFormat object.
func NewQPixelFormat() *QPixelFormat {
	g := newQPixelFormat(QPixelFormat_new())
	g.isSubclass = true
	return g
}

// NewQPixelFormat2 constructs a new QPixelFormat object.
func NewQPixelFormat2(colorModel ColorModel, firstSize byte, secondSize byte, thirdSize byte, fourthSize byte, fifthSize byte, alphaSize byte, alphaUsage AlphaUsage, alphaPosition AlphaPosition, premultiplied AlphaPremultiplied, typeInterpretation TypeInterpretation) *QPixelFormat {
	g := newQPixelFormat(QPixelFormat_new2(colorModel, (uchar)(firstSize), (uchar)(secondSize), (uchar)(thirdSize), (uchar)(fourthSize), (uchar)(fifthSize), (uchar)(alphaSize), alphaUsage, alphaPosition, premultiplied, typeInterpretation))
	g.isSubclass = true
	return g
}

// NewQPixelFormat3 constructs a new QPixelFormat object.
func NewQPixelFormat3(param1 *QPixelFormat) *QPixelFormat {
	g := newQPixelFormat(QPixelFormat_new3(param1.cPointer()))
	g.isSubclass = true
	return g
}

// NewQPixelFormat4 constructs a new QPixelFormat object.
func NewQPixelFormat4(colorModel ColorModel, firstSize byte, secondSize byte, thirdSize byte, fourthSize byte, fifthSize byte, alphaSize byte, alphaUsage AlphaUsage, alphaPosition AlphaPosition, premultiplied AlphaPremultiplied, typeInterpretation TypeInterpretation, byteOrder ByteOrder) *QPixelFormat {
	g := newQPixelFormat(QPixelFormat_new4(colorModel, (uchar)(firstSize), (uchar)(secondSize), (uchar)(thirdSize), (uchar)(fourthSize), (uchar)(fifthSize), (uchar)(alphaSize), alphaUsage, alphaPosition, premultiplied, typeInterpretation, byteOrder))
	g.isSubclass = true
	return g
}

// NewQPixelFormat5 constructs a new QPixelFormat object.
func NewQPixelFormat5(colorModel ColorModel, firstSize byte, secondSize byte, thirdSize byte, fourthSize byte, fifthSize byte, alphaSize byte, alphaUsage AlphaUsage, alphaPosition AlphaPosition, premultiplied AlphaPremultiplied, typeInterpretation TypeInterpretation, byteOrder ByteOrder, subEnum byte) *QPixelFormat {
	g := newQPixelFormat(QPixelFormat_new5(colorModel, (uchar)(firstSize), (uchar)(secondSize), (uchar)(thirdSize), (uchar)(fourthSize), (uchar)(fifthSize), (uchar)(alphaSize), alphaUsage, alphaPosition, premultiplied, typeInterpretation, byteOrder, (uchar)(subEnum)))
	g.isSubclass = true
	return g
}

func (this *QPixelFormat) ColorModel() ColorModel {
	xxxxxxxxx
}

func (this *QPixelFormat) ChannelCount() byte {
	return (byte)(QPixelFormat_ChannelCount(this.h))
}

func (this *QPixelFormat) RedSize() byte {
	return (byte)(QPixelFormat_RedSize(this.h))
}

func (this *QPixelFormat) GreenSize() byte {
	return (byte)(QPixelFormat_GreenSize(this.h))
}

func (this *QPixelFormat) BlueSize() byte {
	return (byte)(QPixelFormat_BlueSize(this.h))
}

func (this *QPixelFormat) CyanSize() byte {
	return (byte)(QPixelFormat_CyanSize(this.h))
}

func (this *QPixelFormat) MagentaSize() byte {
	return (byte)(QPixelFormat_MagentaSize(this.h))
}

func (this *QPixelFormat) YellowSize() byte {
	return (byte)(QPixelFormat_YellowSize(this.h))
}

func (this *QPixelFormat) BlackSize() byte {
	return (byte)(QPixelFormat_BlackSize(this.h))
}

func (this *QPixelFormat) HueSize() byte {
	return (byte)(QPixelFormat_HueSize(this.h))
}

func (this *QPixelFormat) SaturationSize() byte {
	return (byte)(QPixelFormat_SaturationSize(this.h))
}

func (this *QPixelFormat) LightnessSize() byte {
	return (byte)(QPixelFormat_LightnessSize(this.h))
}

func (this *QPixelFormat) BrightnessSize() byte {
	return (byte)(QPixelFormat_BrightnessSize(this.h))
}

func (this *QPixelFormat) AlphaSize() byte {
	return (byte)(QPixelFormat_AlphaSize(this.h))
}

func (this *QPixelFormat) BitsPerPixel() byte {
	return (byte)(QPixelFormat_BitsPerPixel(this.h))
}

func (this *QPixelFormat) AlphaUsage() AlphaUsage {
	xxxxxxxxx
}

func (this *QPixelFormat) AlphaPosition() AlphaPosition {
	xxxxxxxxx
}

func (this *QPixelFormat) Premultiplied() AlphaPremultiplied {
	xxxxxxxxx
}

func (this *QPixelFormat) TypeInterpretation() TypeInterpretation {
	xxxxxxxxx
}

func (this *QPixelFormat) ByteOrder() ByteOrder {
	xxxxxxxxx
}

func (this *QPixelFormat) YuvLayout() YUVLayout {
	xxxxxxxxx
}

func (this *QPixelFormat) SubEnum() byte {
	return (byte)(QPixelFormat_SubEnum(this.h))
}
